src/vmprof_common.h

```diff
@@ -108,9 +114,52 @@ static int opened_profile(char *interp_name, int memory)
  
 PyThreadState* get_current_thread_state(void)
 {
+    PyThreadState *tstate;
 #if PY_MAJOR_VERSION >= 3
-    return (PyThreadState*)_Py_atomic_load_relaxed(&_PyThreadState_Current);
+    tstate = (PyThreadState*)_Py_atomic_load_relaxed(&_PyThreadState_Current);
 #else
-    return _PyThreadState_Current;
+    tstate = _PyThreadState_Current;
 #endif
+
+    if (tstate == NULL) {
```

Comment:
This change is massive, unrelated to the pull request and potentially problematic. The problem here is that we don't hold any lock while walking the interpreter state - that means that if we have more than one thread, it will likely explode. Note that cpython has a lock to do such manipulations, but it's simply not exposed.